File: LLM-Translations/GPT-4/codenet/C/Go/s375567191.go

package main

import (
	"fmt"
)

const MAX = 101

var (
	time            int
	n               int
	g               [MAX][MAX]int
	d, f, color     [MAX]int
	WHITE, GRAY, BLACK = 0, 1, 2
)

func visit(u int) {
	color[u] = GRAY
	d[u] = time + 1
	time++

	for i := 0; i < n; i++ {
		if g[u][i] == 0 {
			continue
		}
		if color[i] == WHITE {
			visit(i)
		}
	}

	color[u] = BLACK
	f[u] = time + 1
	time++
}

func dfs() {
	time = 0

	for i := 0; i < n; i++ {
		if color[i] == WHITE {
			visit(i)
		}
	}

	for i := 0; i < n; i++ {
		fmt.Printf("%d %d %d\n", i+1, d[i], f[i])
	}
}

func main() {
	var x, y, z int

	fmt.Scan(&n)

	for i := 0; i < n; i++ {
		fmt.Scan(&x, &y)
		for j := 0; j < y; j++ {
			fmt.Scan(&z)
			g[x-1][z-1] = 1
		}
	}

	dfs()
}

File: LLM-Translations/GPT-4/codenet/C/Java/s375567191.java
import java.util.Scanner;

public class s375567191 {
    static final int MAX = 101;
    static final int WHITE = 0;
    static final int BLACK = 2;
    static final int GRAY = 1;
    static int time, n;
    static int[][] g = new int[MAX][MAX];
    static int[] d = new int[MAX];
    static int[] f = new int[MAX];
    static int[] color = new int[MAX];

    static void visit(int u) {
        int i;

        color[u] = GRAY;
        d[u] = ++time;

        for (i = 0; i < n; i++) {
            if (g[u][i] == 0)
                continue;
            if (color[i] == WHITE)
                visit(i);
        }

        color[u] = BLACK;
        f[u] = ++time;
    }

    static void dfs() {
        int i;
        time = 0;

        for (i = 0; i < n; i++)
            if (color[i] == WHITE)
                visit(i);

        for (i = 0; i < n; i++)
            System.out.printf("%d %d %d%n", i + 1, d[i], f[i]);
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int i, j, x, y, z;

        n = sc.nextInt();

        for (i = 0; i < n; i++) {
            x = sc.nextInt();
            y = sc.nextInt();
            for (j = 0; j < y; j++) {
                z = sc.nextInt();
                g[x - 1][z - 1] = 1;
            }
        }

        dfs();
    }
}
// 
